sdk/core/core-rest-pipeline/src/pipeline.ts

```diff
@@ -120,8 +120,8 @@ class HttpPipeline implements Pipeline {
   private _policies: PipelineDescriptor[] = [];
   private _orderedPolicies?: PipelinePolicy[];
 
-  private constructor(policies: PipelineDescriptor[] = []) {
-    this._policies = policies;
+  private constructor(policies?: PipelineDescriptor[]) {
+    this._policies = policies?.slice(0) ?? [];
```

Comment:
I am curious, is using `slice` better than spreading? e.g. faster?

Reply:
You'd have to check with a microbenchmark and we know the limits of those. 😉  

I think at one point `slice` was faster just because runtimes had long ago optimized for it, but it's probably not a stretch to imagine they realized people were using spread for this now and optimized that too.